token-lending/program/tests/deposit_obligation_collateral.rs

```diff
@@ -0,0 +1,140 @@
+#![cfg(feature = "test-bpf")]
+
+mod helpers;
+
+use helpers::*;
+use solana_program_test::*;
+use solana_sdk::{
+    pubkey::Pubkey,
+    signature::{Keypair, Signer},
+    transaction::Transaction,
+};
+use spl_token::instruction::approve;
+use spl_token_lending::{
+    instruction::deposit_obligation_collateral, math::Decimal, processor::process_instruction,
+    state::INITIAL_COLLATERAL_RATIO,
+};
+
+const LAMPORTS_TO_SOL: u64 = 1_000_000_000;
+
+#[tokio::test]
+async fn test_success() {
+    let mut test = ProgramTest::new(
+        "spl_token_lending",
+        spl_token_lending::id(),
+        processor!(process_instruction),
+    );
+
+    // limit to track compute unit increase
+    test.set_bpf_compute_max_units(25_000);
+
+    const DEPOSIT_COLLATERAL: u64 = 1 * LAMPORTS_TO_SOL;
+    const DEPOSIT_LIQUIDITY: u64 = DEPOSIT_COLLATERAL / INITIAL_COLLATERAL_RATIO;
+
+    let user_accounts_owner = Keypair::new();
+    let user_transfer_authority = Keypair::new();
+    let usdc_mint = add_usdc_mint(&mut test);
+    let lending_market = add_lending_market(&mut test, usdc_mint.pubkey);
+
+    let sol_reserve = add_reserve(
+        &mut test,
+        &user_accounts_owner,
+        &lending_market,
+        AddReserveArgs {
+            liquidity_amount: DEPOSIT_LIQUIDITY,
+            liquidity_mint_pubkey: spl_token::native_mint::id(),
+            liquidity_mint_decimals: 9,
+            config: TEST_RESERVE_CONFIG,
+            ..AddReserveArgs::default()
+        },
+    );
+
+    let usdc_reserve = add_reserve(
+        &mut test,
+        &user_accounts_owner,
+        &lending_market,
+        AddReserveArgs {
+            liquidity_mint_pubkey: usdc_mint.pubkey,
+            liquidity_mint_decimals: usdc_mint.decimals,
+            config: TEST_RESERVE_CONFIG,
+            ..AddReserveArgs::default()
+        },
+    );
+
+    let obligation = add_obligation(
+        &mut test,
+        &user_accounts_owner,
+        &lending_market,
+        AddObligationArgs {
+            borrow_reserve: &usdc_reserve,
+            collateral_reserve: &sol_reserve,
+            collateral_amount: 0,
+            borrowed_liquidity_wads: Decimal::zero(),
+        },
+    );
+
+    let (mut banks_client, payer, recent_blockhash) = test.start().await;
+
+    let initial_collateral_supply_balance =
+        get_token_balance(&mut banks_client, sol_reserve.collateral_supply).await;
+    let initial_user_collateral_balance =
+        get_token_balance(&mut banks_client, sol_reserve.user_collateral_account).await;
+    let initial_obligation_token_balance =
+        get_token_balance(&mut banks_client, obligation.token_account).await;
```

Comment:
Could you please assert that these initial balances are greater than 0? The base case of 0 is probably worth testing too but we expect deposit collateral to be used for non-zero obligations

Reply:
I changed this test in 1a0f35d8385549c0eae00e4926608f489247d026 to use the same (nonzero) obligation collateral and liquidity balances as the withdraw test. Does it still make sense to assert the balances since they aren't the function being tested?

Reply:
It's sometimes nice to enforce the setup for what *is* being tested, in this case I don't think it matters